src/router.js

```diff
@@ -117,24 +117,33 @@ export function Router(props) {
 	// was the most recent render successful (did not suspend):
 	const didSuspend = useRef();
 	didSuspend.current = false;
-
-	useMemo(() => {
-		// This hack prevents Preact from diffing when we swap `cur` to `prev`:
-		if (this.__v && this.__v.__k) this.__v.__k.reverse();
-
-		count.current++;
-		prev.current = cur.current;
-	}, [url]);
+	// has the route component changed
+	const routeChanged = useRef(false);
 
 	let pr, d, m;
 	toChildArray(props.children).some(vnode => {
 		const matches = exec(rest, vnode.props.path, (m = { ...vnode.props, path: rest, query, params, rest: '' }));
 		if (matches) return (pr = cloneElement(vnode, m));
 		if (vnode.props.default) d = cloneElement(vnode, m);
 	});
+
+	let incoming = pr || d;
```

Comment:
We should really go for better naming in this library and build it instead 😅 the `pr` and `d` stuff never fails to confuse me

Reply:
Every time I need to delve in here I have to relearn what in the world our variables are meant to stand for 

Totally agree